perses/tests/utils.py

```diff
@@ -959,14 +958,22 @@ def validate_endstate_energies_point(htf, endstate=0, minimize=False):
 
     # Check that each of the valence force energies are concordant
     # TODO: Instead of checking with np.isclose(), check whether the ratio of differences is less than a specified energy threshold (like in validate_endstate_energies())
-    for i in range(3):
-        print(f"{components_other[i][0]} -- og: {components_other[i][1]}, hybrid: {components_hybrid[i][1]}")
-        assert np.isclose(components_other[i][1], components_hybrid[i][1])
-
-    # Check that the nonbonded force energies are concordant
+    # Build map between other and rest force keys - keys are for other, values are for rest
+    bonded_keys_other_to_hybrid = {'HarmonicBondForce': 'CustomBondForce', 'HarmonicAngleForce': 'CustomAngleForce',
+                                   'PeriodicTorsionForce': 'CustomTorsionForce'}
```

Comment:
While accessing the force/energy components by name should make it more robust and it doesn't expect any specific order with the `getForces()` method in `openmm`, I have the feeling this is still a bit prone to errors, especially if we start adding multiple `CustomBondForce` objects, which would not be included in the highlighted map. Our current tests/HTFs do not have this, but it is something that may happen in the future.

Reply:
On way is to actually store both the name (from `force.getName()`) and the class name (from `force.__class__.__name__`). But then this means that the part where it validates the energy needs to specify the indices if it wants the force name or the object's class name, which can get confusing. There's "no free lunch".

Reply:
I like that test, but it still will run into problem if we have non-standard names for `Bond` forces, because the only way I could figure out to easily check the bonded energies is using specific/hard-coded names. I agree this might not be a big issue. Will be making the changes and push them to this PR. Thanks!

Reply:
Hmm, how about we just assert that the force names are what we expect them to be before or after creating the dictionary that maps the forces? That way, the test will catch instances where the force names changed and will not even attempt to validate the endstate energies if that is the case.

Something like: 
```python
expected_hybrid_system_force_names = ['CustomBondForce', 'CustomAngleForce', 'CustomTorsionForce', 'CustomNonbondedForce_electrostatics', 'CustomNonbondedForce_sterics', 'CustomBondForce_exceptions', 'NonbondedForce_reciprocal', 'NonbondedForce_sterics']
hybrid_system_force_names = [force.getName() for force in hybrid_system.getForces()]
try: # Remove barostat, if present. If the system is in vacuum phase, it won't have one of these, so we shouldn't require that its present
    hybrid_system_force_names.remove('MonteCarloBarostat')

assert sorted(hybrid_system_force_names) == expected_hybrid_force_names, f"Force names in the RESTCapableHybridTopologyFactory hybrid system do not match the expected force names. Expected: {expected_force_names}, Got: {hybrid_system_force_names}"
```
You could probably do something similar for the `other_system` forces as well.

Reply:
Yes, that may work, but I wanted to avoid doing that, because I don't really have access to the system at that point but to the context, so I would first have to get the system and then the forces. Which is okay, just inconvenient, I guess.

Reply:
Oh wait, we already have the system and a `force_dict`! I think this might work. Will make the changes and let you know.

Reply:
I thought about it and decided to just add the comment, it would add a bit of complexity to do this and I think it is basically analogous to what we already have and are already testing. Maybe we should try to make a test that checks for unique names in the HTFs, but I am gonna leave that for another set of changes, since it seems a bit out of the scope for this one.